### ISSUE TITLE
Optimize DependencyGraph usage

### ISSUE THREAD

**Issue by StjerneIdioten** [StjerneIdioten/I3D-Blender-Addon#57]
**Describe the bug**
Back when the code for exporting meshes was first implemented, blender 2.8 was still very new and the documentation for their new usage of dp's(dependencygraphs) was very sparse.

Their [current documentation](https://docs.blender.org/api/current/bpy.types.Depsgraph.html) is much more thorough and presents a few possibilities for optimizing the usage of the dp's during export. Especially how the dp is currently created from scratch for each shape object exported and the way that the addon doesn't pull from the dependency graph when selecting export objects.

Relates a bit to #56

**Comment by StjerneIdioten** [comment]
After some further bugs have arisen and some testing. I have concluded that something is wrong with blenders dependency graph, or at least the option `preserve_all_data_layers=True` of the `obj.to_mesh()` function.
I did some searching and came across [this](https://developer.blender.org/T79965) active bug on the blender issue tracker.

I have implemented a workaround in 208d80fb7a025be561f5b73993331c0e6140f490 untill blender fixes this bug.
